core/src/main/java/nl/nn/adapterframework/pipes/WsdlXmlValidator.java

```diff
@@ -197,8 +195,44 @@ public void configure() throws ConfigurationException {
 		super.configure();
 	}
 
+	@Override
+	protected PipeForward validate(Message messageToValidate, PipeLineSession session, boolean responseMode, String messageRoot) throws XmlValidatorException, PipeRunException, ConfigurationException {
+		String soapAction = session.get(SoapBindingConstants.SOAP_ACTION, "");
+		if(StringUtils.isNotEmpty(soapAction)) {
+			String soapBodyFromSoapAction = getSoapBodyFromSoapAction(soapAction);
+			if(StringUtils.compare(getSoapBody(), soapBodyFromSoapAction) != 0) {
+				log.debug("soapBody ["+soapBodyFromSoapAction+"] is determined from soapAction ["+soapAction+"]");
+				setSoapBody(soapBodyFromSoapAction);
+				resetCalculatedRootValidatons();
+				configure();
```

Comment:
This is not thread safe.
Isn't it just possible to call super.validate() with messageRoot = soapBodyFromSoapAction?